src/display/graphics.c

```diff
@@ -0,0 +1,387 @@
+	mutex_give(_display_mutex);
+}
+
+uint32_t display_get_color_bg() {
+	mutex_take(_display_mutex, TIMEOUT_MAX);
+	uint32_t color = vexDisplayBackgroundColorGet();
+	mutex_give(_display_mutex);
+	return color;
+}
+
+void display_erase(void) {
+	mutex_take(_display_mutex, TIMEOUT_MAX);
+	vexDisplayErase();
+	mutex_give(_display_mutex);
+}
+
+void display_scroll(int16_t start_line, int16_t lines) {
+	mutex_take(_display_mutex, TIMEOUT_MAX);
+	vexDisplayScroll(start_line, lines);
+	mutex_give(_display_mutex);
+}
+void display_scroll_rect(int16_t x0, int16_t y0, int16_t x1, int16_t y1, int16_t lines) {
+	mutex_take(_display_mutex, TIMEOUT_MAX);
+	vexDisplayScrollRect(x0, y0, x1, y1, lines);
+	mutex_give(_display_mutex);
+}
+
+void display_copy_rect(int16_t x0, int16_t y0, int16_t x1, int16_t y1, uint32_t* buf, int32_t stride) {
+	mutex_take(_display_mutex, TIMEOUT_MAX);
+	vexDisplayCopyRect(x0, y0, x1, y1, buf, stride);
+	mutex_give(_display_mutex);
+}
+
+void display_set_pixel_fg(int16_t x, int16_t y) {
+	mutex_take(_display_mutex, TIMEOUT_MAX);
+	vexDisplayPixelSet(x, y);
+	mutex_give(_display_mutex);
+}
+void display_set_pixel_bg(int16_t x, int16_t y) {
+	mutex_take(_display_mutex, TIMEOUT_MAX);
+	vexDisplayPixelClear(x, y);
+	mutex_give(_display_mutex);
+}
+void display_draw_line(int16_t x0, int16_t y0, int16_t x1, int16_t y1) {
+	mutex_take(_display_mutex, TIMEOUT_MAX);
+	vexDisplayLineDraw(x0, y0, x1, y1);
+	mutex_give(_display_mutex);
+}
+void display_clear_line(int16_t x0, int16_t y0, int16_t x1, int16_t y1) {
+	mutex_take(_display_mutex, TIMEOUT_MAX);
+	vexDisplayLineClear(x0, y0, x1, y1);
+	mutex_give(_display_mutex);
+}
+
+void display_draw_rect(int16_t x0, int16_t y0, int16_t x1, int16_t y1) {
+	mutex_take(_display_mutex, TIMEOUT_MAX);
+	vexDisplayRectDraw(x0, y0, x1, y1);
+	mutex_give(_display_mutex);
+}
+void display_clear_rect(int16_t x0, int16_t y0, int16_t x1, int16_t y1) {
+	mutex_take(_display_mutex, TIMEOUT_MAX);
+	vexDisplayRectClear(x0, y0, x1, y1);
+	mutex_give(_display_mutex);
+}
+void display_fill_rect(int16_t x0, int16_t y0, int16_t x1, int16_t y1) {
+	mutex_take(_display_mutex, TIMEOUT_MAX);
+	vexDisplayRectFill(x0, y0, x1, y1);
+	mutex_give(_display_mutex);
+}
+
+void display_draw_circle(int16_t x, int16_t y, int16_t r) {
+	mutex_take(_display_mutex, TIMEOUT_MAX);
+	vexDisplayCircleDraw(x, y, r);
+	mutex_give(_display_mutex);
+}
+void display_clear_circle(int16_t x, int16_t y, int16_t r) {
+	mutex_take(_display_mutex, TIMEOUT_MAX);
+	vexDisplayCircleClear(x, y, r);
+	mutex_give(_display_mutex);
+}
+void display_fill_circle(int16_t x, int16_t y, int16_t r) {
+	mutex_take(_display_mutex, TIMEOUT_MAX);
+	vexDisplayCircleFill(x, y, r);
+	mutex_give(_display_mutex);
+}
+
+//Text Display Functions
+void display_vprintf(uint8_t txt_fmt, const int16_t line, const char* fmt, va_list args){
+    mutex_take(_display_mutex, TIMEOUT_MAX);
+	char* out;
+	vasprintf(&out, fmt, args);
+	va_list empty;
+	switch(txt_fmt){
+		case 0:
```

Comment:
Looks like you've got a mixture of tabs and spaces here.

Reply:
This was attempted to be fixed... but it doesn't seem to work? I'm starting to think this is an editor quirk rather than what we thought it was.

Reply:
Nevermind I did something pretty jank to get git to recognize the change :/